libraries/AP_Frsky_Telem/AP_Frsky_Telem.cpp

```diff
@@ -682,6 +682,13 @@ uint32_t AP_Frsky_Telem::calc_batt(uint8_t instance)
 uint32_t AP_Frsky_Telem::calc_ap_status(void)
 {
     uint32_t ap_status;
+    
+    // prepare imu temp to a 6 bit value with a range from 20° - 83° value offset is 19
+    // 1 = 20°, 63 = 82°
+    uint32_t imutemp = AP::ins().get_temperature(0) - 19;
+    if (imutemp <= 0) imutemp = 0;
+    if (imutemp >= 63) imutemp = 63;
```

Comment:
Have you tested your conversion routine alone in a compiler? Have you tested with temperatures of 18°C or less?

Using a standalone compiler to test your conversions, I found the following:
- A "get_temperature" of 18°C or less will end up giving a value of 63 to imutemp...
- Also there's an issue with rounding of the float returned by get_temperature(): for example, a "get_temperature" of 19.9°C will end up giving a value of 0 to imutemp (when it should really be rounded to 1), and as another example a "get_temperature" of 81.5°C will end up giving a value of 62 to imutemp (when it should really be rounded to 63).

There appears to be no need to create a variable imutemp (anything helps I guess to optimize memory). All can fit on one line (even if it may not look quite as readable). I propose adding to AP_Frsky_Telem.h the following:
`#define AP_IMU_TEMP_ORIGIN          19`
and replacing your code with this one line:
`ap_status =| ((AP::ins().get_temperature(0) < AP_IMU_TEMP_ORIGIN) ? 0 : (AP::ins().get_temperature(0) > (AP_IMU_TEMP_ORIGIN + AP_IMU_TEMP_LIMIT)) ? AP_IMU_TEMP_LIMIT : (uint8_t)roundf(AP::ins().get_temperature(0) - AP_IMU_TEMP_ORIGIN))<<AP_IMU_TEMP_OFFSET;`
I tested this routine in a standalone compiler with "get_temperatures" values such as 18.5, 19, 19.4, 19.5, 20, 81.4, 81.5, 82, etc. and it all seems to round up nicely... Let me know what you think.